### PR TITLE
feat: add function to calculate VMT share for each day of the year

### PR DESCRIPTION
**Pull request by danielolsen**
### Purpose
Add a function to calculate VMT share for each day of the year.

### What the code is doing
Two data files are added, `moves_daily.csv` and `moves_monthly.csv`, which contain exactly the data as it's presented in the report on MOVES. The 2010 version of the report is used in the Attribution file since it contains the monthly distributions for both regular years and leap years.

A helper function is added which takes a year and an area type (either `"urban"` or `"rural"`) and returns a pandas Series where the index is each day of the year and the values are the fraction of VMT that are estimated to occur in that day.

### Testing
Tested manually for leap years and regular years, and for urban vs. rural. Values are spot checked against the previous values, which are relevant for a regular year that begins on a Sunday, and values for other combinations are checked to ensure that the year sums to one (subject to floating point precision).

### Usage Example/Visuals
```python
>>> from prereise.gather.demanddata.transportation_electrification.data_helper import generate_daily_weighting
>>> generate_daily_weighting(2017)
2017-01-01    0.001964
2017-01-02    0.002520
2017-01-03    0.002520
2017-01-04    0.002520
2017-01-05    0.002520
                ...
2017-12-27    0.002786
2017-12-28    0.002786
2017-12-29    0.002786
2017-12-30    0.002171
2017-12-31    0.002171
Freq: D, Length: 365, dtype: float64
>>> generate_daily_weighting(2017).sum()
0.9999999999999999
>>> generate_daily_weighting(2020)
2020-01-01    0.002494
2020-01-02    0.002494
2020-01-03    0.002494
2020-01-04    0.001943
2020-01-05    0.001943
                ...
2020-12-27    0.002132
2020-12-28    0.002737
2020-12-29    0.002737
2020-12-30    0.002737
2020-12-31    0.002737
Freq: D, Length: 366, dtype: float64
>>> generate_daily_weighting(2020).sum()
1.0
>>> generate_daily_weighting(2017, "rural")
2017-01-01    0.002302
2017-01-02    0.002381
2017-01-03    0.002381
2017-01-04    0.002381
2017-01-05    0.002381
                ...
2017-12-27    0.002616
2017-12-28    0.002616
2017-12-29    0.002616
2017-12-30    0.002528
2017-12-31    0.002528
Freq: D, Length: 365, dtype: float64
>>> generate_daily_weighting(2017, "rural").sum()
0.9999999999999999
>>> generate_daily_weighting(2017, "suburban")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\DanielOlsen\repos\bes\PreREISE\prereise\gather\demanddata\transportation_electrification\data_helper.py", line 107, in generate_daily_weighting
    raise ValueError(f"area_type must be one of {allowable_area_types}")
ValueError: area_type must be one of {'rural', 'urban'}
```

### Time estimate
15 minutes.

### CODE REVIEW

File: prereise/gather/demanddata/transportation_electrification/data_helper.py [hunk 1]
```diff
@@ -89,3 +92,39 @@ def update_if_weekend(data: pd.DataFrame):
...
+    data_dir = os.path.join(os.path.dirname(__file__), "data")
+    monthly_distribution = pd.read_csv(
+        os.path.join(data_dir, "moves_monthly.csv"), index_col=0
+    )
+    weekday_distribution = pd.read_csv(
+        os.path.join(data_dir, "moves_daily.csv"), index_col=0
+    )
+    weekday_distribution.loc["weekday"] /= 5  # normalize to each day within weekdays
+    weekday_distribution.loc["weekend"] /= 2  # normalize to each day within weekends
+    year_type = "leap_year" if calendar.isleap(int(year)) else "regular_year"
```
Comment (BainanXia) [inline prereise/gather/demanddata/transportation_electrification/data_helper.py:118]: Nice, never use `calendar` before.

File: prereise/gather/demanddata/transportation_electrification/data_helper.py [hunk 2]
```diff
@@ -89,3 +92,39 @@ def update_if_weekend(data: pd.DataFrame):
     data.loc[data["Day of Week"].isin(range(2, 7)), "If Weekend"] = 2
 
     return data
+
+
+def generate_daily_weighting(year, area_type="urban"):
+    """Generate daily weighting factors based on vehicle-miles-travelled distributions.
+
+    :param int/str year: year to generate weighting factors for.
+    :param str area_type: Either 'urban' or 'rural'.
```
Comment (BainanXia) [inline prereise/gather/demanddata/transportation_electrification/data_helper.py:101]: We will need to add the doc string for the error raised here.

**Review by BainanXia — APPROVED** [review]
LGTM

File: prereise/gather/demanddata/transportation_electrification/data_helper.py (hunk 2 above)
Comment (danielolsen) [reply prereise/gather/demanddata/transportation_electrification/data_helper.py:101]: Done.

**Review by rouille — APPROVED** [review]
🙏

**Review by jenhagg — APPROVED** [review]

**Pull request merged**